02-express-tutorial/controllers/people.js

```diff
@@ -0,0 +1,74 @@
+let { people } = require("../data");
+
+const getPeople = (req, res) => {
+  console.log("Inside getAllPeople");
+  res.status(200).json({ success: true, data: people });
+};
+
+const getPerson = (req, res) => {
+  console.log(`inside getPerson with ${req.params.id}`);
+  const id = parseInt(req.params.id);
+  const person = people.find((person) => person.id === id);
+  if (!person) {
+    return res.status(404).json({ message: "There is no person with that id" });
+  }
+  return res.status(200).json(person);
+};
+
+const getMaxId = () => {
+  return people.reduce((ret, cur) => (ret < cur.id ? cur.id : ret), 0);
+};
+
+const addPerson = (req, res) => {
+  console.log(req.body);
+  const { name } = req.body;
+  if (!name) {
+    return res
+      .status(400)
+      .json({ success: false, message: "Please provide a name" });
+  }
+
+  newId = getMaxId() + 1;
+  people.push({ id: newId, name: req.body.name });
+  res.status(201).json({ success: true, name: name, id: newId });
```

Comment:
Requires change:
Instead of looking through all of the id's to find the max, you can simply use the `people` array length and add 1 to it. On a related note, make sure you define variables with `let` or `const`, otherwise, they'll be defined as globals.
```js
const newId = people.length + 1;

people.push({ id: newId, name: req.body.name });

res.status(201).json({ success: true, name: name, id: newId });
```